src/engine/strat_engine/backstore/backstore.rs

```diff
@@ -716,6 +713,35 @@ impl Backstore {
     pub fn rename_pool(&mut self, new_name: &Name) -> StratisResult<()> {
         self.data_tier.block_mgr.rename_pool(new_name)
     }
+
+    /// A summary of block sizes
+    pub fn block_size_summary(&self, tier: BlockDevTier) -> Option<BlockSizeSummary> {
+        match tier {
+            BlockDevTier::Data => Some(self.data_tier.partition_by_use().into()),
+            BlockDevTier::Cache => self
+                .cache_tier
+                .as_ref()
+                .map(|ct| ct.partition_cache_by_use().into()),
+        }
+    }
+
+    /// What the pool's action availability should be
+    pub fn action_availability(&self) -> ActionAvailability {
+        let data_tier_bs_summary = self
+            .block_size_summary(BlockDevTier::Data)
+            .expect("always exists");
+        let cache_tier_bs_summary: Option<BlockSizeSummary> =
+            self.block_size_summary(BlockDevTier::Cache);
+        if let Err(err) = data_tier_bs_summary.validate() {
+            warn!("Disabling pool changes for this pool: {}", err);
+            ActionAvailability::NoPoolChanges
+        } else if let Some(Err(err)) = cache_tier_bs_summary.map(|ct| ct.validate()) {
+            warn!("Disabling pool changes for this pool: {}", err);
+            ActionAvailability::NoPoolChanges
+        } else {
```

Comment:
Is this more on principal or am I missing a protection that this provides? My understanding is that the cache always gets greedily allocated so it seems that putting this in `NoPoolChanges` mode for the cache may come a little bit too late as the device would have already been extended to use varying sized blocks. If the cache were lazily allocated, I'd understand this a bit more.

Reply:
I think that this condition will not happen, exactly for the reason you specify. But if it does, then NoPoolChanges would be correct.

Reply:
I've added a note.